mass-deployer/pkg/mass-deployer/deployer.go

```diff
@@ -0,0 +1,281 @@
+
+			network := workloads.ZNet{
+				Name:        fmt.Sprintf("%s%dnetwork", vmGroup.Name, i),
+				Description: "network for mass deployment",
+				Nodes:       []uint32{nodeID},
+				IPRange: gridtypes.NewIPNet(net.IPNet{
+					IP:   net.IPv4(10, 20, 0, 0),
+					Mask: net.CIDRMask(16, 32),
+				}),
+				AddWGAccess: false,
+			}
+			w := workloads.VM{
+				Name:        fmt.Sprintf("%s%d", vmGroup.Name, i),
+				NetworkName: network.Name,
+				Flist:       vmGroup.Flist,
+				CPU:         int(vmGroup.FreeCPU),
+				Memory:      int(vmGroup.FreeMRU),
+				PublicIP:    vmGroup.Pubip4,
+				PublicIP6:   vmGroup.Pubip6,
+				Planetary:   vmGroup.Planetary,
+				RootfsSize:  int(*convertGBToBytes(vmGroup.Rootsize)),
+				Entrypoint:  vmGroup.Entrypoint,
+				EnvVars:     map[string]string{"SSH_KEY": sshKeys[vmGroup.SSHKey]},
+				Mounts:      mounts,
+			}
+			deployment := workloads.NewDeployment(w.Name, nodeID, "", nil, network.Name, disks, nil, []workloads.VM{w}, nil)
+
+			vmDeployments = append(vmDeployments, &deployment)
+			networkDeployments = append(networkDeployments, &network)
+			deploymentsInfo = append(deploymentsInfo, vmDeploymentInfo{nodeID: nodeID, deploymentName: deployment.Name, vmName: w.Name})
+		}
+	}
+	return groupDeploymentsInfo{vmDeployments: vmDeployments, networkDeployments: networkDeployments, deploymentsInfo: deploymentsInfo}
+}
+
+func loadDeploymentsInfo(tfPluginClient deployer.TFPluginClient, deployments []vmDeploymentInfo) []string {
+	vmsInfo := []string{}
+	var lock sync.Mutex
+	var wg sync.WaitGroup
+
+	for _, info := range deployments {
+		wg.Add(1)
+
+		go func(depInfo vmDeploymentInfo) {
+			defer wg.Done()
+
+			vm, err := tfPluginClient.State.LoadVMFromGrid(depInfo.nodeID, depInfo.vmName, depInfo.deploymentName)
+			if err != nil {
+				log.Debug().Err(err).Msgf("couldn't load vm %s of deployment %s from node %d", depInfo.vmName, depInfo.deploymentName, depInfo.nodeID)
+				return
+			}
+
+			vmInfo := struct {
+				Name      string
+				PublicIP4 string
+				PublicIP6 string
+				YggIP     string
+				IP        string
+				Mounts    []workloads.Mount
+			}{vm.Name, vm.ComputedIP, vm.ComputedIP6, vm.YggIP, vm.IP, vm.Mounts}
+
+			groupInfo, err := yaml.Marshal(vmInfo)
+			if err != nil {
+				log.Debug().Err(err).Msg("failed to marshal json")
+			}
+
+			lock.Lock()
+			defer lock.Unlock()
+			vmsInfo = append(vmsInfo, string(groupInfo))
+		}(info)
+	}
+
+	wg.Wait()
+	return vmsInfo
+}
+
+func parseDisks(name string, disks []Disk) (disksWorkloads []workloads.Disk, mountsWorkloads []workloads.Mount) {
+	for _, disk := range disks {
+		DiskWorkload := workloads.Disk{
+			Name:   fmt.Sprintf("%sdisk", name),
+			SizeGB: int(disk.Size),
+		}
+
+		disksWorkloads = append(disksWorkloads, DiskWorkload)
+		mountsWorkloads = append(mountsWorkloads, workloads.Mount{DiskName: DiskWorkload.Name, MountPoint: disk.Mount})
+	}
+	return
+}
+
+func convertGBToBytes(gb uint64) *uint64 {
```

Comment:
Why pointer to uint64? it's either 0 or you got a value to pass no?

Reply:
it's a pointer to make it easy [here](https://github.com/threefoldtech/tfgrid-sdk-go/pull/527/files#diff-c4ca62f5c85b79fde950680322cb4a2735539c7a947aa63471ce63ee07bacd27R95-R102) to use the returned value directly in the filter, instead of saving the returned value in a variable then using it